File: algorithm/278. First Bad Version/278.cpp
// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int tem,start=0,end=n;
        while(end-start>1)
        {
            tem=start+(end-start)/2;
            if(isBadVersion(tem))
            {
                end=tem;
            }
            else
            {
                start=tem;
            }
            
        }
        return end;
    }
};

File: algorithm/278. First Bad Version/278.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution(object):
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        tem,start,end=0,0,n
        while end-start>1:
            tem=start+(end-start)/2
            if isBadVersion(tem):
                end=tem
            else:
                start=tem
        return end
